src/amplitude_experiment/deployment/deployment_runner.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from typing import Optional, Set
+import threading
+
+from ..local.config import LocalEvaluationConfig
+from ..cohort.cohort_loader import CohortLoader
+from ..cohort.cohort_storage import CohortStorage
+from ..flag.flag_config_api import FlagConfigApi
+from ..flag.flag_config_storage import FlagConfigStorage
+from ..local.poller import Poller
+from ..util.flag_config import get_all_cohort_ids_from_flag
+
+
+class DeploymentRunner:
+    def __init__(
+            self,
+            config: LocalEvaluationConfig,
+            flag_config_api: FlagConfigApi,
+            flag_config_storage: FlagConfigStorage,
+            cohort_storage: CohortStorage,
+            cohort_loader: Optional[CohortLoader] = None,
+    ):
+        self.config = config
+        self.flag_config_api = flag_config_api
+        self.flag_config_storage = flag_config_storage
+        self.cohort_storage = cohort_storage
+        self.cohort_loader = cohort_loader
+        self.lock = threading.Lock()
+        self.poller = Poller(self.config.flag_config_polling_interval_millis / 1000, self.__periodic_refresh)
+        self.logger = logging.getLogger("Amplitude")
+        self.logger.addHandler(logging.StreamHandler())
+        if self.config.debug:
+            self.logger.setLevel(logging.DEBUG)
+
+    def start(self):
+        with self.lock:
+            self.refresh(initial=True)
+            self.poller.start()
+
+    def stop(self):
+        self.poller.stop()
+
+    def __periodic_refresh(self):
+        try:
+            self.refresh()
+        except Exception as e:
+            self.logger.error(f"Refresh flag and cohort configs failed: {e}")
+
+    def refresh(self, initial: bool = False):
```

Comment:
remove `initial` and always throw an exception on error. If calling inside the poller then handle the exception and continue polling.